### ISSUE TITLE
Subtle argument ordering issue in HEAD

### ISSUE THREAD

**Issue by amilsted** [Jutho/LinearMaps.jl#7]
Hi Jutho!

I noticed that the position of the element type argument in the FunctionMap version of LinearMap has changed since the last release. Unfortunately the previous ordering, although incorrect, is still a valid call!

```julia
f!(dest::AbstractVector{Complex128}, s_in::AbstractVector{Complex128}) = begin
    copy!(dest, s_in)
end

op1 = LinearMap(f!, 10, Complex128)
@show eltype(op1)

op2 = LinearMap(f!, Complex128, 10)
@show eltype(op2)

@show op1
@show op2
```

In the second case, which worked in version 0.2.0, `fc` gets set to `Complex128`.

**Comment by Jutho** [comment]
Hi Ashley,

thanks for reporting. I guess this is on master? I'll try to free some time later this week to fix this. I have some other updates in the pipeline.

Best,

Jutho

**Comment by Jutho** [comment]
I think this is now fixed. I also tagged a new version which should be approved later today.

Next version (some updates soon) will fix julia 0.6 deprecations but become 0.6 (and higher) only. Let me know if there are any other other requests from the PI julia users.

**Comment by amilsted** [comment]
Thanks a lot @Jutho, your work is appreciated! I'll let you know if we discover anything else.
